src/kvstore.h

```diff
@@ -70,6 +70,12 @@ typedef dict *(kvstoreDictLUTDefragFunction)(dict *d);
 void kvstoreDictLUTDefrag(kvstore *kvs, kvstoreDictLUTDefragFunction *defragfn);
 void *kvstoreDictFetchValue(kvstore *kvs, int didx, const void *key);
 dictEntry *kvstoreDictFind(kvstore *kvs, int didx, void *key);
+/*
+ * The kvstore handles `key` based on `dictType` during initialization:
+ * - If `dictType.embedded-entry` is 1, it clones the `key`.
+ * - Otherwise, it assumes ownership of the `key`.
+ * The caller must ensure the `key` is properly freed.
+ */
 dictEntry *kvstoreDictAddRaw(kvstore *kvs, int didx, void *key, dictEntry **existing);
```

Comment:
Function docs are in the .c files in Valkey (not sure why, but it's consistently so).

I'm not completely sure about the kvstore abstraction, but I think we should mention in which cases each of these are used. The dictType is an implementation detail the caller of kvstore functions shouldn't need to worry about. For example say that for database keys (`db->keys`), it copies the key and for `db->expires` and sharded pubsub channels, it doesn't.

Reply:
Moving some of the content to kvstoreDictCreate and adding the different usage(s) here.